src/openms/source/ANALYSIS/QUANTITATION/TMTEighteenPlexQuantitationMethod.cpp

```diff
@@ -0,0 +1,219 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Timo Sachsenberg $
+// $Authors: Stephan Aiche, Radu Suciu $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/ANALYSIS/QUANTITATION/TMTEighteenPlexQuantitationMethod.h>
+
+#include <OpenMS/DATASTRUCTURES/ListUtils.h>
+#include <OpenMS/DATASTRUCTURES/Matrix.h>
+
+
+#include <algorithm>
+
+
+namespace OpenMS
+{
+const String TMTEighteenPlexQuantitationMethod::name_ = "tmt18plex";
+const std::vector<std::string> TMTEighteenPlexQuantitationMethod::channel_names_ = {"126","127N","127C","128N","128C","129N","129C","130N","130C","131N","131C","132N","132C","133N","133C","134N","134C","135N"};
+
+TMTEighteenPlexQuantitationMethod::TMTEighteenPlexQuantitationMethod()
+{
+    setName("TMTEighteenPlexQuantitationMethod");
+
+    //    // mass map outline - for further details please see #2427 (was adapted for tmt18plex)
+    //    "126", 126.127726, x, x, 127C, 128C
+    //    "127N", 127.124761, x, x, 128N, 129N
+    //    "127C", 127.131081, x, 126, 128C, 129C
+    //    "128N", 128.128116, x, 127N, 129N, 130N
+    //    "128C", 128.134436, 126, 127C, 129C, 130C
+    //    "129N", 129.131471, 127N, 128N, 130N, 131N
+    //    "129C", 129.137790, 127C, 128C, 130C, 131C
+    //    "130N", 130.134825, 128N, 129N, 131N, x
+    //    "130C", 130.141145, 128C, 129C, 131C, x
+    //    "131N", 131.138180, 129N, 130N, x, x
+    //    "131C", 131.144500, 129C, 130C, x, x
+
+    // create the channel map                                                //-2  -1  +1  +2
+    // TODO: double check this
+    channels_.push_back(IsobaricChannelInformation("126",   0, "", 126.127726, -1, -1,  2,  4));
+    channels_.push_back(IsobaricChannelInformation("127N",  1, "", 127.124761, -1, -1,  3,  5));
+    channels_.push_back(IsobaricChannelInformation("127C",  2, "", 127.131081, -1,  0,  4,  6));
+    channels_.push_back(IsobaricChannelInformation("128N",  3, "", 128.128116, -1,  1,  5,  7));
+    channels_.push_back(IsobaricChannelInformation("128C",  4, "", 128.134436,  -1,  2,  6,  8));
+    channels_.push_back(IsobaricChannelInformation("129N",  5, "", 129.131471,  -1,  3,  7,  9)));
```

Comment:
```suggestion
    channels_.push_back(IsobaricChannelInformation("128C",  4, "", 128.134436,  0,  2,  6,  8));
    channels_.push_back(IsobaricChannelInformation("129N",  5, "", 129.131471,  1,  3,  7,  9)));
```
The -2 channels are 0% in the example sheet (so filling in a `-1` serves the same purpose), but another sheet could contain value >0%, thus we should use the mapping.